src/glyph/builder.rs

```diff
@@ -268,107 +268,145 @@ impl GlyphBuilder {
 pub struct OutlineBuilder {
     identifiers: HashSet<Identifier>,
     outline: Outline,
-    scratch_contour: Option<Contour>,
-    number_of_offcurves: u32,
+    scratch_state: OutlineBuilderState,
+}
+
+#[derive(Debug)]
+enum OutlineBuilderState {
+    Idle,
+    Drawing { scratch_contour: Contour, number_of_offcurves: u32 },
+}
+
+impl Default for OutlineBuilderState {
+    fn default() -> Self {
+        OutlineBuilderState::Idle
+    }
 }
 
 impl OutlineBuilder {
     pub fn new() -> Self {
         Default::default()
     }
 
-    /// Start a new path to be added to the glyph with `end_path()`.
+    /// Begin a new path to be added to the glyph.
+    ///
+    /// It must be finished with [`Self::end_path`] before the outline can be [`Self::finish`]ed
+    /// and retrieved.
     ///
     /// Errors when:
     /// 1. a path has been begun already but not ended yet.
     /// 2. the identifier is not unique within the glyph.
+    ///
+    /// On error, it won't begin a new path and you can continue drawing the previously started
+    /// path.
```

Comment:
This feels iffy. Shouldn't calls to `start_path` that are not the *first* path in an outline always be preceded with `end_path`?

Reply:
Yes; the error case is that you do `begin_path` followed by `begin_path`. As in, this errors out if you didn't do `end_path` after the first `begin_path`.